tracim/views/core_api/schemas.py

```diff
@@ -355,3 +382,129 @@ class ContentDigestSchema(marshmallow.Schema):
                     'for sub-contents. Default is True. '
                     'In first version of the API, this field is always True',
     )
+
+
+#####
+# Content
+#####
+
+class ContentSchema(ContentDigestSchema):
+    current_revision_id = marshmallow.fields.Int(example=12)
+    created = marshmallow.fields.DateTime(
+        format='%Y-%m-%dT%H:%M:%SZ',
+        description='Content creation date',
+    )
+    author = marshmallow.fields.Nested(UserDigestSchema)
+    modified = marshmallow.fields.DateTime(
+        format='%Y-%m-%dT%H:%M:%SZ',
+        description='date of last modification of content',
+    )
+    last_modifier = marshmallow.fields.Nested(UserDigestSchema)
+
+
+class ThreadContentSchema(ContentSchema):
+    content_type = marshmallow.fields.Str(
+        example='thread',
+        validate=OneOf([content.slug for content in CONTENT_DEFAULT_TYPE]),
+    )
+    raw_content = marshmallow.fields.String('Description of Thread')
+
+
+class HtmlDocumentContentSchema(ContentSchema):
+    content_type = marshmallow.fields.Str(
+        example='html-content',
+        validate=OneOf([content.slug for content in CONTENT_DEFAULT_TYPE]),
+    )
+    raw_content = marshmallow.fields.String('<p>Html page Content !</p>')
+
+#####
+# Revision
+#####
+
+
+class RevisionSchema(ContentDigestSchema):
+    comments_ids = marshmallow.fields.List(marshmallow.fields.Int(example=4))
+    revision_id = marshmallow.fields.Int(example=12)
+    created = marshmallow.fields.DateTime(
+        format='%Y-%m-%dT%H:%M:%SZ',
+        description='Content creation date',
+    )
+    author = marshmallow.fields.Nested(UserDigestSchema)
+
+
+class ThreadRevisionSchema(RevisionSchema):
+    content_type = marshmallow.fields.Str(
+        example='thread',
+        validate=OneOf([content.slug for content in CONTENT_DEFAULT_TYPE]),
+    )
+    raw_content = marshmallow.fields.String('Description of Thread')
+
+
+class HtmlDocumentRevisionSchema(RevisionSchema):
+    content_type = marshmallow.fields.Str(
+        example='html-content',
+        validate=OneOf([content.slug for content in CONTENT_DEFAULT_TYPE]),
+    )
+    raw_content = marshmallow.fields.String('<p>Html page Content !</p>')
+
+
+####
+
+class CommentSchema(marshmallow.Schema):
+    content_id = marshmallow.fields.Int(example=6)
+    parent_id = marshmallow.fields.Int(example=34)
+    raw_content = marshmallow.fields.String(
+        example='<p>This is just an html comment !</p>'
+    )
+    author = marshmallow.fields.Nested(UserDigestSchema)
+    created = marshmallow.fields.DateTime(
+        format='%Y-%m-%dT%H:%M:%SZ',
+        description='comment creation date',
+    )
+
+
+class ContentModifySchema(marshmallow.Schema):
+    label = marshmallow.fields.String(
+        example='contract for client XXX',
+        description='New title of the content'
+    )
+
+
+class HtmlDocumentModifySchema(ContentModifySchema):
```

Comment:
This schema is to rename to `TextBasedContentModifySchema` and to use in place of `HTMLDocumentUpdate` and `ThreadUpdate`

Reply:
to be done in same time as https://github.com/tracim/tracim/issues/618